src/controller/OperationalCredentialsDelegate.h

```diff
@@ -28,6 +29,14 @@
 namespace chip {
 namespace Controller {
 
+typedef void (*NOCGenerated)(void * context, const ByteSpan & noc, const PeerId & deviceId);
+
+// TODO - Reduce memory requirement for generating x509 certificates
+// As per specifications (section 6.3.7. Trusted Root CA Certificates), DER certs
+// should require 600 bytes. Currently, due to ASN writer overheads, a larger buffer
+// is needed, even though the generated certificate fits in 600 bytes limit.
+constexpr uint32_t kMaxCHIPDERCertLength = 1024;
```

Comment:
So, if there is computational overhead to generate the cert, it should be accounted separately from the `kMaxCHIPDERCertLength`. The `kMaxCHIPDERCertLength` should still be 600, and when you allocate, you can do `kMaxCHIPDERCerLength + kCertGenerationOverhead` or something like that.

Reply:
Fwiw, with https://github.com/project-chip/connectedhomeip/pull/7832 merged I think we should be able to set 600 here...

Reply:
Updated to 600 bytes.